
Created by Przemysław 'Troll' Maciąg <pmaciag (at) gmail (dot) com>

In cvs for 0.9.3 version somebody broke's trayicon...
This patch restores ability to show proper trayicon
in FVWM and few others (light) WM's.

diff -u -p -r1.24 -r1.26
--- src/tools/trayicon/trayicon_x11.cpp	2004/02/28 16:52:39	1.24
+++ src/tools/trayicon/trayicon_x11.cpp	2004/08/20 03:22:01	1.26
@@ -251,11 +261,6 @@ TrayIconFreeDesktop::TrayIconFreeDesktop
 
 	if ( manager_window != None )
 		send_message(dsp, manager_window, SYSTEM_TRAY_REQUEST_DOCK, winId(), 0, 0);
-	else
-	{
-		object->hide();
-		return;
-	}
 
 	// some KDE mumbo-jumbo... why is it there? anybody?
 	Atom kwm_dockwindow_atom = XInternAtom(dsp, "KWM_DOCKWINDOW", false);
@@ -345,9 +371,7 @@ void TrayIcon::sysInstall()
 		d = (TrayIconPrivate *)(new TrayIconFreeDesktop(this, pm));
 
 	sysUpdateToolTip();
+	d->show();
-
-	if ( v_isWMDock )
-		d->show();
 }
 
 void TrayIcon::sysRemove()
